src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerServices/FunctionPointerOps.cs

```diff
@@ -51,7 +51,7 @@ public override int GetHashCode()
 
         private static uint s_genericFunctionPointerNextIndex;
         private const uint c_genericDictionaryChunkSize = 1024;
-        private static LowLevelList<IntPtr> s_genericFunctionPointerCollection = new LowLevelList<IntPtr>();
+        private static readonly List<IntPtr> s_genericFunctionPointerCollection = new List<IntPtr>();
```

Comment:
This `s_genericFunctionPointerCollection` pool looks like a premature optimization. I think it would be fine to allocate `GenericMethodDescriptorInfo`s one at a time.

Reply:
`s_genericFunctionPointerCollection` does not enable any `GenericMethodDescriptor` reuse. There is `s_genericFunctionPointerDictionary` for that. I am not suggesting deleting `s_genericFunctionPointerDictionary`.

`s_genericFunctionPointerCollection` reduces frequency of `NativeMemory.Alloc` calls. We allocate spaces for `c_genericDictionaryChunkSize` `GenericMethodDescriptor`s at a time using `NativeMemory.Alloc` and then hand them out one at a time.